### PR TITLE
fix(Datatable): fix selection using old selected state

### PR DESCRIPTION
**Pull request by ajkl2533**

### CODE REVIEW

File: src/components/Datatable/Datatable.tsx [hunk 1]
```diff
@@ -172,20 +176,31 @@ const Datatable = <D extends Record<string, unknown>>({
     [onRowsSelect],
   );
 
+  const memoizedTableConfig = useDeepCompareMemo(
```
Comment (PrincAm) [inline src/components/Datatable/Datatable.tsx:179]: Good optimization! What is the problematic dependency, `restTableConfig`?

Comment (ajkl2533) [reply src/components/Datatable/Datatable.tsx:179]: It's one object, so that was the primary issue

**Comment by youssefssc** [comment]
is it my setup or is select/deselect actually slow?

File: src/components/Datatable/Datatable.tsx [hunk 2]
```diff
@@ -172,20 +176,31 @@ const Datatable = <D extends Record<string, unknown>>({
...
+        value={{
+          totalLength: totalDataSize,
+          selectedIds,
+          selectedLength: selectedIds.length,
+          defaultHiddenColumns,
+          defaultColumnOrder,
+          hasExclusionLogic,
+          setHasExclusionLogic,
+          hasSelection,
+          setShouldResetSelectedRows,
```
Comment (PrincAm) [inline src/components/Datatable/Datatable.tsx:201]: Pls, who is the consumer of `setShouldResetSelectedRows`?

**Review by PrincAm — APPROVED** [review]
LGTM, hope it works as expected 😉

File: src/components/Datatable/Datatable.tsx (hunk 1 above)
Comment (PrincAm) [reply src/components/Datatable/Datatable.tsx:179]: Yep, I wasn't sure if same handler fn also wasn't causing unnecessary call

File: src/components/Datatable/Datatable.tsx (hunk 2 above)
Comment (ajkl2533) [reply src/components/Datatable/Datatable.tsx:201]: The consumer is `ElementCounter`, this handles situation when you have some selected items and choose `Select none` from the dropdown menu. It wasn't working before, now the `Table` component gets notified through this prop that it should reset selection

**Comment by ajkl2533** [comment]
@youssefssc nope it's slow as hell, I'll work on it later, we need to stabilize functionality now and I'll try to fix performance when we have time

**Review by youssefssc — APPROVED** [review]

File: src/components/Datatable/Datatable.tsx (hunk 2 above)
Comment (PrincAm) [reply src/components/Datatable/Datatable.tsx:201]: Right, I missed `ElementCounter`, see it now. Thanks for explanation

**Pull request merged**